### PR TITLE
Sign UserOps using SessionKey

### PR DESCRIPTION
**Pull request by kaushalrajbacancy**
## Description

- Added `signUserOpWithSessionKey` function into **SessionKeyValidator** for **sign the userOP** using session key.
- Added `sessionData` function into SessionKeyValidator to fetch session Data of particular **sessionKey**.
- Updated `send` function as now it has one optional `isUserOpAlreadySigned`(boolean) parameter to know if userOp already signed or not.
- Updated `enableSessionKey`, `rotateSessionKey` and `disableSessionKey` functions for the **multiple session keys** per apiKey and walletAddress.

## Types of changes
What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Further comments (optional)

- Install module: https://sepolia.etherscan.io/tx/0x018165e631a39d3119f27e08e86c8cb4ac91d7be0f67dcb4e635a992648fc35a
- Enable sessionKey: https://sepolia.etherscan.io/tx/0x75294f31e06b52ea25fcbb977ae86f037c6157dc6e6b74dd15b29df39a639444
- Rotate sessionKey : https://sepolia.etherscan.io/tx/0xb8500fe71fbaf0fc6a13f4297129505849e1dc2742f6f5fc9b0679fdccedfd54
- Disable sessionKey: https://sepolia.etherscan.io/tx/0x9b5349923fbc945827f33bb378561ebe024eb986f88c8c0eefd2c10a206bc4b5
- Sent USDC with signed signature using sessionKey: https://sepolia.etherscan.io/tx/0x59ac65395381dc56e60e379913572afbc17bca63938297d71c87e8f158a61874

### CODE REVIEW

File: src/sdk/SessionKeyValidator/SessionKeyValidator.ts [hunk 1]
```diff
@@ -334,6 +358,38 @@ export class SessionKeyValidator {
         }
     }
 
+    private async getSignUserOpHash(
```
Comment (ch4r10t33r) [inline src/sdk/SessionKeyValidator/SessionKeyValidator.ts:361]: I wouldn't expose this in the SDK. I'll handle all of this in the backend.

File: src/sdk/SessionKeyValidator/SessionKeyValidator.ts [hunk 2]
```diff
@@ -173,6 +174,29 @@ export class SessionKeyValidator {
         }
     }
 
+    async signUserOpHash(userOpHash: string, userOp: UserOperation): Promise<UserOperation> {
```
Comment (ch4r10t33r) [inline src/sdk/SessionKeyValidator/SessionKeyValidator.ts:177]: Refactor and rename this as `signUserOpWithSessionKey`, There can be multiple active session keys per `apiKey` and `walletAddress`. This scenario doesn't handle it.

**Review by ch4r10t33r — CHANGES_REQUESTED** [review]

File: examples/03-transfer-erc20-session-key.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+async function main() {
+  // initializating sdk...
+  const modularSdk = new ModularSdk({ privateKey: process.env.WALLET_PRIVATE_KEY }, { chainId: Number(process.env.CHAIN_ID), bundlerProvider: new EtherspotBundler(Number(process.env.CHAIN_ID), bundlerApiKey) })
+
+  const sessionKeyModule = new SessionKeyValidator(
+    modularSdk,
+    new EtherspotBundler(Number(process.env.CHAIN_ID), bundlerApiKey)
+  )
+
+  console.log('address: ', modularSdk.state.EOAAddress)
```
[the remark this answers is not in the document]
Comment (kaushalrajbacancy) [reply examples/03-transfer-erc20-session-key.ts]: Removed

File: src/sdk/SessionKeyValidator/SessionKeyValidator.ts (hunk 1 above)
Comment (kaushalrajbacancy) [reply src/sdk/SessionKeyValidator/SessionKeyValidator.ts]: Changes done.

File: src/sdk/SessionKeyValidator/SessionKeyValidator.ts (hunk 2 above)
Comment (kaushalrajbacancy) [reply src/sdk/SessionKeyValidator/SessionKeyValidator.ts]: Changes done.

File: examples/03-transfer-erc20-session-key.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+  // clear the transaction batch
+  await modularSdk.clearUserOpsFromBatch();
+
+  // add transactions to the batch
+  const userOpsBatch = await modularSdk.addUserOpsToBatch({ to: tokenAddress, data: transactionData });
+  console.log('transactions: ', userOpsBatch);
+
+  // estimate transactions added to the batch and get the fee data for the UserOp
+  const op = await modularSdk.estimate({
```
[the remark this answers is not in the document]
Comment (kaushalrajbacancy) [reply examples/03-transfer-erc20-session-key.ts:47]: Changes done.

**Review by ch4r10t33r — APPROVED** [review]
LGTM

**Pull request merged**